src/java/org/apache/cassandra/auth/CassandraAuthorizer.java

```diff
@@ -168,6 +193,8 @@ public void revokeAllOn(IResource droppedResource)
                                                       ClientState.forInternalCalls()));
 
             executeLoggedBatch(statements);
+
+            invalidateRolesAndPermissions(roles, droppedResource);
```

Comment:
Also, I may be wrong, but verify this scenario - role A has granted a role B, role B has granted a permission on X.

In database, role B has permission on X, but A does not. However, in the cache both A has permission on X and B has permission on X.

In this particular case , we will pass only role B for invalidation. We will find the relationship between A and B in roles cache and that relationship will be removed - however if we ask the cache whether A has permission on X, it will reply with true I suppose. 

----

No I can see it should work because PermissionsCache does not take the "user" argument into account - which is confusing but makes this mechanism work - maybe then remove that unused argument?

Reply:
Yes ... I went though several iterations with that.

I did find it necessary to invalidate all entries by resource, as I think your example shows too. I will remove the unused "user" argument.